src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -313,14 +293,33 @@ case class WdlBinding(ast: Ast, source: WdlSource, importResolver: ImportResolve
     }
     val taskName = sourceString(ast.getAttribute("task"))
     val task = findTask(taskName) getOrElse invalidTask
-    val inputs = WdlBinding.getCallInput(ast)
+    val inputs = processCallInput(ast)
     new Call(alias, taskName, task, inputs.toMap, this)
   }
 
+  private def processCallInput(ast: Ast): Map[String, WdlExpression] = AstTools.getCallInputAsts(ast).map {a =>
+    val key = sourceString(a.getAttribute("key"))
+    val expression = new WdlExpression(a.getAttribute("value"))
+    (key, expression)
+  }.toMap
+
+  private def processWdlType(ast: AstNode): WdlType = {
+    ast match {
+      case t: Terminal if t.getSourceString == WdlFileType.toWdlString => WdlFileType
+      case t: Terminal if t.getSourceString == WdlStringType.toWdlString => WdlStringType
+      case t: Terminal if t.getSourceString == WdlIntegerType.toWdlString => WdlIntegerType
+      case t: Terminal if t.getSourceString == WdlFloatType.toWdlString => WdlFloatType
+      case t: Terminal if t.getSourceString == WdlBooleanType.toWdlString => WdlBooleanType
+      case t: Terminal if t.getSourceString == WdlObjectType.toWdlString => WdlObjectType
```

Comment:
I'm not going to do battle over this but if this is an exhaustive search over the potential cases my preference would be to have a function which takes a `Terminal` and outputs a `WdlType`, and then a single line in the `match` which calls it when `ast` is a `Termina`l. The concept of converting a `Terminal` to a `WdlType` seems separate to me from the concept of converting an `AstNode` to a `WdlType`.

Or perhaps have a method on `Terminal` which returns the appropriate `WdlType`

And if it's possible to boil that down you might be able to convert this whole thing to something like this:

    Option(ast).getOrElse(WdlStringType).collect({case t: Terminal => t.toWdlType}).headOption.getOrElse(throw new ....)

I don't think even I would advocate doing that all in a single line, and I don't know I'd advocate doing it that way but just an off the cuff example. I think since this is effectively 3-way logic my ideal would be a 3-line match though.

I've only looked at this code block and haven't explored around, so it's entirely possible there are realities which make neither of these feasible